src/basic/namespace-util.c

```diff
@@ -236,3 +236,31 @@ int userns_acquire(const char *uid_map, const char *gid_map) {
         return TAKE_FD(userns_fd);
 
 }
+
+int in_same_namespace(pid_t pid1, pid_t pid2, NamespaceType type) {
+        const char *ns_path;
+        struct stat ns_st1, ns_st2;
+
+        if (pid1 == pid2)
+                return 1;
```

Comment:
This should be moved to after the two `getpid_cached()`. As, pid1 may be non-zero but the current PID, and pid2 may be zero.

Reply:
done